internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -627,7 +619,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	for i, cd := range res.Composed {
 		// Specifying a name for P&T templates is optional but encouraged.
 		// If there was no name, fall back to using the index.
-		id := cd.ResourceName
+		id := string(cd.ResourceName)
```

Comment:
I like to do this because it makes it obvious what the string map key is. e.g.:

```go
m: = map[ResourceName]Resource{}

// vs

n := map[string]Resource{}
```